services/cd-service/pkg/service/batch.go

```diff
@@ -93,11 +93,17 @@ func ValidateApplication(
 }
 
 func (d *BatchServer) processAction(
-	batchAction *api.BatchAction,
+	batchAction *api.BatchAction, user *auth.User,
 ) (repository.Transformer, *api.BatchResult, error) {
 	switch action := batchAction.Action.(type) {
 	case *api.BatchAction_CreateEnvironmentLock:
 		act := action.CreateEnvironmentLock
+		if d.RBACConfig.DexEnabled {
```

Comment:
I would avoid inlining this whole check and maybe just have an iterface that you call to authorize this call.